tests/test_datasets/test_dataset.py

```diff
@@ -99,6 +107,58 @@ def test_get_data_with_ignore_attributes(self):
         self.assertEqual(len(categorical), 38)
         # TODO test multiple ignore attributes!
 
+    def test_get_data_numpy(self):
+        data = self.titanic.get_data(dataset_format='array')
+        self.assertTrue(isinstance(data, np.ndarray))
+        self.assertEqual(data.shape[1], len(self.titanic.features))
+        self.assertEqual(data.shape[0], 1309)
+        self.assertTrue(data.dtype == 'object')
+
+        X, y = self.titanic.get_data(
+            dataset_format='array',
+            target=self.titanic.default_target_attribute)
+        self.assertTrue(isinstance(X, np.ndarray))
+        self.assertTrue(isinstance(y, np.ndarray))
+        self.assertEqual(X.shape, (1309, 13))
+        self.assertEqual(y.shape, (1309,))
+        self.assertTrue(X.dtype == 'object')
+        self.assertTrue(y.dtype == int)
+
+    def test_get_data_pandas(self):
+        data = self.titanic.get_data(dataset_format='dataframe')
+        self.assertTrue(isinstance(data, pd.DataFrame))
+        self.assertEqual(data.shape[1], len(self.titanic.features))
+        self.assertEqual(data.shape[0], 1309)
+        col_dtype = {
+            'pclass': 'float64',
+            'survived': 'category',
+            'name': 'object',
+            'sex': 'category',
+            'age': 'float64',
+            'sibsp': 'float64',
+            'parch': 'float64',
+            'ticket': 'object',
+            'fare': 'float64',
+            'cabin': 'object',
+            'embarked': 'category',
+            'boat': 'object',
```

Comment:
Now that I'm seeing this, I don't think that casting integer columns with missing values to object is in the spirit of arff. The latest version mentioned that float and int should be both treated as numerical (see [here](https://github.com/Waikato/weka-3.8/blob/master/wekadocs/manual/arff.tex)), so I think we should do the same here.

Reply:
Does Weka cast integer to float when reading an integer column?

Reply:
I don't think so: https://github.com/Waikato/weka-3.8/blob/49865490cef763855ede07cd11331a7aeaecd110/weka/src/main/java/weka/core/converters/ArffLoader.java#L694

Also, reading this arff file:
```
@relation in-vs-float

@attribute attr1 real
@attribute attr2 integer
@attribute attr3 numeric

@data
1, 2, 3
1.0, 2.0, 3.0
1.1, 1.0, 1.1
```
results in the following being displayed in WEKA:
![image](https://user-images.githubusercontent.com/5320498/49740189-fecc6400-fc93-11e8-86db-b05c5c69448c.png)

Reply:
and what happens with 

```
@relation in-vs-float

@attribute attr1 real
@attribute attr2 integer
@attribute attr3 numeric

@data
1, 2, 3
1.0, 2, 3.0
1.1, 1, 1.1
```

Reply:
I am really wondering what is the point of putting some header then :)
So I would remove the type inference and only keep the unpacking of the categories then.

Reply:
> I am really wondering what is the point of putting some header then :)

I stopped wondering about arff...

> So I would remove the type inference and only keep the unpacking of the categories then.

That should be sufficient.

Reply:
We are fine here then.